### PR TITLE
Bugtool enhancement and small fixes

### PR DESCRIPTION
**Pull request by aanm**
**Summary of changes**:

- Add missing copyright headers
- Add 5 second timeout for all commands executed
- Lift root permissions for kubernetes commands
- Add timestamp to the generated tar file
- Compress base directory instead of uppermost directory

### CODE REVIEW

File: bugtool/cmd/root.go [hunk 1]
```diff
@@ -27,12 +42,16 @@ var BugtoolRootCmd = &cobra.Command{
 	},
 }
 
-const disclaimer = `DISCLAIMER
+const (
+	disclaimer = `DISCLAIMER
 This tool has copied information about your environment.
 If you are going to register a issue on GitHub, please
```
Comment (joestringer) [inline bugtool/cmd/root.go:49]: while we're doing misc fixups, can update the grammar on this to say `an issue`.

File: bugtool/cmd/root.go [hunk 2]
```diff
@@ -56,14 +75,17 @@ func init() {
 func runTool() {
 	// Search for a Cilium pod and if one is found prefix all of the
 	// commands to use the pod(s).
-	k8sPods, _ = getCiliumPods(k8sNamespace, k8sLabel)
-	if len(k8sPods) == 0 {
-		common.RequireRootPrivilege("bugtool")
+	var err error
+	k8sPods, err = getCiliumPods(k8sNamespace, k8sLabel)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "....Failed to retrieve cilium logs from kubernetes: %s\n", err)
```
Comment (joestringer) [inline bugtool/cmd/root.go]: Does this always happen even if you run in an environment without kubernetes?

**Review by joestringer — COMMENTED** [review]
Looks good to me, minor comments.

**Comment by aalemayhu** [comment]
Why are you adding a 5-second timeout? Would be good to either wait for filtering support or have at least a higher default timeout like 90-second.

File: bugtool/cmd/root.go (hunk 2 above)
Comment (aalemayhu) [reply bugtool/cmd/root.go]: The root requirement is not for Kubernetes but for the case where we detect no pods and would like to run all the commands on the host. In that case, we need root possibly for some commands and copying the state directory or we see something like

```
Could not copy state directory open /var/run/cilium/state: permission denied
```

**Review by aalemayhu — CHANGES_REQUESTED** [review]
Most of it looks good to me but would change the timeout and do something different with the root check. Since you are changing the default behaviour. Would prefer to not assume we are running Kubernetes by default and make the timeout configurable.

**Comment by aanm** [comment]
>Most of it looks good to me but would change the timeout and do something different with the root check. Since you are changing the default behaviour. Would prefer to not assume we are running Kubernetes by default and make the timeout configurable.

I can change the timeout to be customizable but the execTimeout was mostly because it was taking too much time for an environment where kubectl was not being run.

I remove the root requirement because it was to run the bugtool for a system that was didn't have kubernetes running. It doesn't make sense to require root because we can't reach kube-apiserver.

IMO, we should not require the users to run a program with root privileges without tell them why do we actually need the privileges.

File: bugtool/cmd/root.go (hunk 2 above)
Comment (joestringer) [reply bugtool/cmd/root.go]: What I meant to ask about is this: if there's no kubernetes available then presumably `getCiliumPods()` will always fail, so users of the tool would always see this error message. However, that error message would make no sense to report in a non-kubernetes environment.

**Comment by tgraf** [comment]
We should add a clear info message when we drop out of Kubernetes mode. It's fine to require root privileges then but we should print a message to indicate why that is required.

Other changes LGTM

File: bugtool/cmd/root.go (hunk 1 above)
Comment (aanm) [reply bugtool/cmd/root.go:49]: Done

**Comment by aalemayhu** [comment]
Honestly, I did not think about the cases where `kube-apiserver` might be unavailable. If you have a better idea on how to detect if we have pods, please share. Maybe the user should have to specify the mode and that might help with providing the correct error message. Wdyt? If it's better for you we could also drop this root check to merge what is already good and then discuss the mode detection and error messages.

**Comment by tgraf** [comment]
> Honestly, I did not think about the cases where kube-apiserver might be unavailable. If you have a better idea on how to detect if we have pods, please share. Maybe the user should have to specify the mode and that might help with providing the correct error message. Wdyt? If it's better for you we could also drop this root check to merge what is already good and then discuss the mode detection and error messages.

I suggest that we print the error but in a way that makes it obvious:

```
Unable to contact Kubernetes API server, falling back to local information retrieval...
```

In that case, require root privileges.

**Comment by tgraf** [comment]
Added back the root check

**Review by tgraf — APPROVED** [review]

**Pull request merged**